package org.confluence.mod.common.entity.projectile.arrow;

import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.projectile.AbstractArrow;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;
import org.confluence.mod.Confluence;

public class StarArrowEntity extends BaseArrowEntity {
    private static final ResourceLocation TEXTURE = Confluence.asResource("textures/entity/arrow/star_arrow.png");
    private static final ResourceLocation PARTICLE = Confluence.asResource("falling_star");

    public StarArrowEntity(EntityType<? extends AbstractArrow> entityType, Level level) {
        super(entityType, level);
    }

    public StarArrowEntity(EntityType<? extends AbstractArrow> entityType, LivingEntity owner, ItemStack pickupItemStack, ItemStack firedFromWeapon) {
        super(entityType, owner, pickupItemStack, firedFromWeapon);
    }
    @Override
    protected int getPenetrationCount() {
        return 99;
    }

    @Override
    protected double getAdditionalKnockback() {
        return 2.0;
    }

    @Override
    public void shoot(double x, double y, double z, float velocity, float inaccuracy) {
        super.shoot(x, y, z, velocity, inaccuracy);
        setDeltaMovement(getDeltaMovement().scale(0.8));
    }

    @Override
    public double getDefaultGravity() {
        return 0;
    }

    @Override
    protected int getAutoDiscardTick() {
        return 50;
    }

    @Override
    protected int getLuminance() {
        return 8;
    }

    @Override
    protected ResourceLocation getParticleId() {
        return PARTICLE;
    }

    @Override
    public ResourceLocation getTexturePath() {
        return TEXTURE;
    }
}
